ml_pipeline_engine/dag/manager.py

```diff
@@ -671,36 +710,29 @@ async def _run_recurrent_subgraph(self, node_id: NodeId, node_result: Recurrent,
         self._node_storage.set_active_rec_subgraph(start_from_node_id, node_id)
         max_iterations = self.dag.graph.nodes[node_id].get(NodeField.max_iterations)
 
-        logger.debug(
-            'Start the process of the recurrent subgraph for nodes start_node=%s, dest_node=%s',
-            start_from_node_id,
-            node_id,
-        )
-
         recurrent_subgraph = get_connected_subgraph(
-            self.dag.graph, start_from_node_id, node_id, is_recurrent=True,
+            self.dag.graph, start_from_node_id, node_id, is_recurrent=True, is_oneof=dag.is_oneof,
```

Comment:
question, since we started to pass a subgraph to the method,
why do we still searching for a  `start_from_node_id` from `self.dag`; and calling `get_connected_subgraph` with  `self.dag.graph` but not `dag.graph`; 
I had an idea that we don't need to use self.dag since we have a reduced dag as input argument (correct me if I'm wrong)

Reply:
Check how the rec subgraph works. dag.graph != recurrent_subgraph